lib/AuroraLip/Texture/Formats/TPL.cs

```diff
@@ -42,7 +42,7 @@ protected override void Read(Stream stream)
             int TotalImageCount = stream.ReadInt32(Endian.Big);
```

Comment:
```
        public virtual bool CanRead => true;

        public virtual bool CanWrite => true;
        
       public virtual bool IsMatch(Stream stream, in string extension = "")
             => Matcher(stream, extension
```

```
        protected override void Read(Stream stream)
        {
            long HeaderStart = stream.Position;

            if (!Matcher(stream))
                throw new InvalidIdentifierException("0x0020AF30");
```

should be a slightly better approach.

Reply:
Yeah, you mentioned virtual but is inheritance really appropriate here for these methods/properties?  For instance, when reading the TPL from the `FIPAFTEX ` we wouldn't want the `IsMatch` calls to be read from thechild (`FIPAFTEX `), we'd want it to be read from the TPL?

Maybe a better way to do this would be to have `FIPAFTEX` _have a_ `TPL` member that it uses to call into.  That would make the relationship clearer.

Reply:
`Read` is `protected`.  Very unfortunate :(.

Just my take but the `TPL` class shouldn't know anything about other classes that might use it, they are not related at all so modifying the `TPL` class to be virtual seems wrong.

Reply:
> Yeah, you mentioned virtual but is inheritance really appropriate here for these methods/properties? For instance, when reading the TPL from the `FIPAFTEX ` we wouldn't want the `IsMatch` calls to be read from thechild (`FIPAFTEX `), we'd want it to be read from the TPL?

This is just what we want, the FIPAFTEX class should not use the `IsMatch` of the TPL class.
what happens when we do this?
```
JUTTexture tex = new FIPAFTEX();
tex.IsMatch(stream)
```
if the steam is a FIPAFTEX file we should get false currently.
for a TPL date it would be true, If we then try to read the file we will get an error.

Reply:
Yeah I see what you mean.  How would you feel if I made `Read` public?

Reply:
you can also make a helper class which is a sub class of the tpl class, your main class can then use JUTTexture, IMagicIdentify and IFileAccess.

For me it is only important that there are no errors, if we ever change the behavior.

Reply:
Ah I like the helper class idea.  I'll do that.  Thanks @Venomalia !

Reply:
This is now addressed!